backend/tests/Feature/VisitsApiTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Entities\Browser;
+use App\Entities\Demographic;
+use App\Entities\Device;
+use App\Entities\GeoPosition;
+use App\Entities\Os;
+use App\Entities\Page;
+use App\Entities\Session;
+use App\Entities\System;
+use App\Entities\User;
+use App\Entities\Visit;
+use App\Entities\Visitor;
+use App\Entities\Website;
+use DateTime;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class VisitsApiTest extends TestCase
+{
+    use RefreshDatabase;
+
+    private $user;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->user = factory(User::class)->create();
+        factory(Website::class)->create();
+        factory(Visitor::class)->create();
+        factory(Page::class)->create();
+        factory(Device::class)->create();
+        factory(Browser::class)->create();
+        factory(Os::class)->create();
+        factory(GeoPosition::class)->create();
+        factory(Demographic::class)->create();
+        factory(System::class)->create();
+        factory(Session::class)->create();
+    }
+
+    public function testPageViewsFilter()
```

Comment:
test looks amazingly, but could you also test cases when visits do not belong to website. Visits out of chosen date range. And case when no visits for chosen period

Reply:
But do not touch current test, it is okay. Add 3 different tests for each case